Algorithms/Numeric/Factorial.cs

```diff
@@ -22,5 +23,35 @@ public static long Calculate(int num)
 
             return num == 0 ? 1 : num * Calculate(num - 1);
         }
+
+        /// <summary>
+        ///     Calculates factorial of a big number.
+        /// </summary>
+        /// <param name="num">Big Input number.</param>
+        /// <returns>Factorial of Big input number.</returns>
+        public static BigInteger Calculate(BigInteger num)
```

Comment:
Could you implement a generic factorial function that would have an INumber constraint? That'd work for any type without duplication

Reply:
1) I can make this method generic to accept all types of INTEGER numbers using IBinaryInteger<T> interface rather than INumber<T>(According to me decimal numbers are out of scope for factorials). Is that what you are looking for?
2) To use INumber or IBinaryInteger project needs to be updated to .NET 7. Is that something you do or I can do it?

Reply:
1. Thanks for correction, `IBinaryInteger` is what I meant
2. Feel free to update projects to .NET 7, but do it in a separate PR. This way it will be easier to review.
Thanks!